Source/GUI/preferences.cpp

```diff
@@ -175,3 +218,76 @@ void Preferences::OnRejected()
     Load();
 }
 
+void Preferences::on_testConnection_pushButton_clicked()
+{
+    struct UI {
+        static void setSuccess(QLabel* label, QPushButton* button) {
+            label->setStyleSheet("color: green");
+            label->setText("Success!");
+            button->setEnabled(true);
+        }
+        static void setError(QLabel* label, const QString& error, QPushButton* button) {
+            label->setStyleSheet("color: red");
+            label->setText(error);
+            button->setEnabled(true);
+        }
+        static void setChecking(QLabel* label, QPushButton* button) {
+            label->setStyleSheet("");
+            label->setText("Checking..");
+            button->setEnabled(false);
+        }
+    };
+
+    UI::setChecking(ui->connectionTest_label, ui->testConnection_pushButton);
+
+    QNetworkAccessManager manager;
+    QNetworkRequest request(ui->signalServerUrl_lineEdit->text() + "/fileuploads/upload/test");
+    request.setRawHeader("Authorization", "Basic " + QByteArray(QString("%1:%2")
+							 .arg(ui->signalServerLogin_lineEdit->text())
+							 .arg(ui->signalServerPassword_lineEdit->text()).toLocal8Bit().toBase64()));
```

Comment:
Agree, authorization: basic is not the best way, but that's what backend provides for now.